### PR TITLE
Add Gatekeeper constraints to the discovered policies table

### PR DESCRIPTION
**Pull request by mprahl**
Relates:
https://issues.redhat.com/browse/ACM-14159

### CODE REVIEW

File: frontend/public/locales/en/translation.json [hunk 1]
```diff
@@ -1161,6 +1161,7 @@
   "Discovered policies": "Discovered policies",
   "discoveredPolicies.tooltip.clusterViolation": "Gatekeeper violations refer to the total number of pre-existing misconfigurations that do not comply with the defined constraints.",
   "discoveredPolicies.tooltip.responseAction": "In Open Cluster Management policies, response action refers to the remediationAction. For Gatekeeper, it refers to the enforcementAction.",
+  "discoveredPolicies.tooltip.severity": "For Gatekeeper constraints, this value comes from the policy.open-cluster-management.io/severity annotation.",
```
Comment (mprahl) [inline frontend/public/locales/en/translation.json:1164]: @dockerymick could you please review this new sentence?

Comment (dockerymick) [reply frontend/public/locales/en/translation.json:1164]: It looks good to me!

Comment (mprahl) [reply frontend/public/locales/en/translation.json:1164]: Thank you!

File: frontend/src/routes/Governance/discovered/ByCluster/DiscoveredByClusterPage.test.tsx [hunk 2]
```diff
@@ -54,7 +55,7 @@ describe('DiscoveredByClusterPage', () => {
               name: 'check-policy-reports',
               namespace: 'managed1',
               compliant: 'Compliant',
-              remediationAction: 'inform',
+              responseAction: 'inform',
```
Comment (yiraeChristineKim) [inline frontend/src/routes/Governance/discovered/ByCluster/DiscoveredByClusterPage.test.tsx:58]: Why did you change this?

File: frontend/src/routes/Governance/common/util.tsx [hunk 3]
```diff
@@ -703,39 +704,46 @@ export function getPolicySource(
...
-export function getEngineString(kind: string): string {
-  switch (kind) {
-    case 'ConfigurationPolicy':
-    case 'CertificatePolicy':
-    case 'OperatorPolicy':
+export function getEngineString(apiGroup: string): string {
+  switch (apiGroup) {
+    case 'policy.open-cluster-management.io':
       return 'Open Cluster Management'
+    case 'constraints.gatekeeper.sh':
```
Comment (yiraeChristineKim) [inline frontend/src/routes/Governance/common/util.tsx:711]: Can you add constraintTempate too? because of this https://github.com/stolostron/policy-collection/blob/6f7996f4c2abdc6da70b950b0176517bb1ad48d2/community/AC-Access-Control/policy-gatekeeper-limitclusteradmin.yaml#L37

File: frontend/src/routes/Governance/discovered/ByCluster/DiscoveredByClusterPage.test.tsx (hunk 2 above)
Comment (mprahl) [reply frontend/src/routes/Governance/discovered/ByCluster/DiscoveredByClusterPage.test.tsx:58]: That way the UI code doesn't have to differentiate between `remediationAction` and `enforcementAction` based on the policy type. The grouping.ts file does that for you.

File: frontend/src/routes/Governance/discovered/ByCluster/common.tsx [hunk 4]
```diff
@@ -36,6 +44,18 @@ export const policyViolationSummary = (discoveredPolicyItems: DiscoveredPolicyIt
...
+export const getConstraintCompliance = (totalViolations?: number): string => {
+  totalViolations = totalViolations || -1
+
+  if (totalViolations === 0) {
+    return 'compliant'
+  } else if (totalViolations > 0) {
+    return 'noncompliant'
+  }
+
+  return ''
```
Comment (yiraeChristineKim) [inline frontend/src/routes/Governance/discovered/ByCluster/common.tsx:56]: '-' for consistency

File: frontend/src/routes/Governance/discovered/grouping.test.ts [hunk 5]
```diff
@@ -250,59 +278,124 @@ describe('OnMessage test', () => {
...
+      [],
+      helmRelease,
+      channels,
+      subscriptions,
+      resolveSource.toString(),
+      getSourceText.toString(),
+      parseStringMap.toString(),
+      parseDiscoveredPolicies.toString()
+    )
+    // All policies name are same
```
Comment (yiraeChristineKim) [inline frontend/src/routes/Governance/discovered/grouping.test.ts:398]: Is this comment correct?

Comment (mprahl) [reply frontend/src/routes/Governance/discovered/grouping.test.ts:398]: Good point, it was a copy/paste error.

File: frontend/src/routes/Governance/discovered/grouping.ts [hunk 6]
```diff
@@ -169,10 +192,26 @@ export function grouping(): {
...
+
+      if (allResponseActions.size === 1) {
+        responseAction = allResponseActionsList[0]
+      } else if (
+        allResponseActions.size === 2 &&
+        allResponseActions.has('inform') &&
+        allResponseActions.has('enforce')
+      ) {
+        responseAction = 'inform/enforce'
+      } else {
```
Comment (yiraeChristineKim) [inline frontend/src/routes/Governance/discovered/grouping.ts:206]: What is this case? any example?

File: frontend/src/routes/Governance/discovered/useFetchPolicies.tsx [hunk 7]
```diff
@@ -19,20 +18,22 @@ export interface DiscoveredPolicyItem {
...
+  compliant?: string
+  responseAction: string
+  severity?: string
   _isExternal: boolean
   annotation: string
   created: string
   label: string
   kind_plural: string
-  namespace: string
+  namespace?: string
```
Comment (yiraeChristineKim) [inline frontend/src/routes/Governance/discovered/useFetchPolicies.tsx:29]: Can add comment here, undefined case is Gatekeeper, similar like this

File: frontend/src/routes/Governance/discovered/grouping.ts (hunk 6 above)
Comment (mprahl) [reply frontend/src/routes/Governance/discovered/grouping.ts:206]: If there are multiple enforcement actions (e.g. `deny` and `warn` becomes `deny/warn` in the UI).

**Comment by yiraeChristineKim** [comment]
I was about to open https://issues.redhat.com/browse/ACM-13333 PR but there are overlaps so I will open after your PR merged Thanks!

File: frontend/src/routes/Governance/discovered/grouping.ts (hunk 6 above)
Comment (yiraeChristineKim) [reply frontend/src/routes/Governance/discovered/grouping.ts:206]: gotcha

File: frontend/src/routes/Governance/discovered/DiscoveredPolicies.tsx [hunk 8]
```diff
@@ -49,7 +49,7 @@ function nameCell(item: DiscoverdPolicyTableItem): ReactNode {
         policyName: item.name,
         apiGroup: item.policies[0].apigroup,
         apiVersion: item.policies[0].apiversion,
-        policyNamespace: item.policies[0].cluster,
+        policyNamespace: null,
```
Comment (yiraeChristineKim) [inline frontend/src/routes/Governance/discovered/DiscoveredPolicies.tsx:52]: Should we remove this in` NavigationPath.discoveredByCluster` look like nowhere to use

Comment (yiraeChristineKim) [reply frontend/src/routes/Governance/discovered/DiscoveredPolicies.tsx:52]: Also `discoveredByCluster` group by apiGroup + kind + name so we don't need this my apolgies

**Review by yiraeChristineKim — APPROVED** [review]
